### PR TITLE
adding the ability to set a defaultValue on metadata field

### LINKED ISSUE #6509: default value on metadata field
Creators can require some metadata to be supplied when checking out.
The team at EthGlobal asked if there was a way to supply a default value in order to skip steps for their users who might have already entered some data in their application.

### PR DESCRIPTION
**Pull request by julien51**
# Description

The team at ethGlobal asked for this ;)

# Issues

Fixes #6509

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have updated the docs to reflect my changes if applicable
- [X] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [X] I have performed a self-review of my own code
- [X] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by julien51** [comment]
image

File: unlock-app/src/components/interface/checkout/MetadataForm.tsx [hunk 1]
```diff
@@ -10,11 +10,21 @@ interface Props {
   onSubmit: (metadata: UserMetadata) => void
 }
 
+interface DefautltValues {
```
Comment (cnasc) [inline unlock-app/src/components/interface/checkout/MetadataForm.tsx:13]:
```suggestion
interface DefaultValues {
```

File: unlock-app/src/unlockTypes.ts [hunk 2]
```diff
@@ -172,6 +172,7 @@ export interface KeyholdersByLock {
 
 export interface MetadataInput {
   name: string
+  defaultValue: string
```
Comment (cnasc) [inline unlock-app/src/unlockTypes.ts:175]: Do we want to require a defaultValue?

Comment (julien51) [reply unlock-app/src/unlockTypes.ts:175]: no, it should be optional. good catch.

**Comment by julien51** [comment]
PTAL @cnasc

**Comment by cnasc** [comment]
I think this looks good overall, but there are 2 test cases we should probably add to https://github.com/unlock-protocol/unlock/blob/master/unlock-app/src/__tests__/components/interface/checkout/MetadataForm.test.tsx

- Verify that the form can be submitted with an unchanged default value
- Verify that a default value can be overwritten by the user when filling out the form

(and maybe while we're in there we can fix the interface name typo)

**Comment by julien51** [comment]
Added PTAL!

**Pull request merged**